src/features/ProductPage/Product/WithImage/WithImage.tsx

```diff
@@ -70,8 +70,11 @@ export const ProductWithImage = ({
   );
 
   useEffect(() => {
-    setSelectedPrice(selectedVariant.prices[0]);
-  }, [selectedVariant]);
+    const price =
+      selectedVariant.prices.find((price) => price.intervalId === selectedPrice.intervalId) ??
+      selectedVariant.prices[0];
+    setSelectedPrice(price);
+  }, [selectedPrice.intervalId, selectedVariant]);
```

Comment:
This works every time. One nitpick is that the interval selection radio button flickers when choosing different sizes:
![2022-09-20 15 04 06](https://user-images.githubusercontent.com/428120/191344164-bc4f66f3-6320-4691-b437-e68f53160351.gif)
This is probably due to using `useEffect` which will fire after a new selection is made. This might be able to be fixed by wrapping `setSelection` or adding this functionality into `useVariantOption`. Adding to `useVariantOption` make sense so you don't need to repeat this `WithImageGrid.tsx`

Reply:
Sure, but this is unrelated to this PR. It has always flickered. I can look at it in another story.

Reply:
Don't bother in this epic. I was unfamiliar with the existing behavior.